### PR TITLE
Rsa suite6 tests

### PR DESCRIPTION
**Pull request by ValentinVaninetti**
##Summary
Adds 6 integration tests for RSA sign/verify, encrypt/decrypt, and error handling.

##Changes
New Tests (6)

> Sign and verify with correct key
> Signature rejection on tampered message (edge case)
> Signature rejection with a different key (edge case)
> Encrypt/decrypt round-trip
> Sign with unknown key returns RsaKeyNotFound
> Decrypt with unknown key returns RsaKeyNotFound

##New file: src/tests/rsa_test.rs Modified: src/tests/mod.rs (registered new module)

### CODE REVIEW

File: src/tests/rsa_test.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    fn random_bytes() -> Vec<u8> {
+        let mut buf = [0u8; 32];
+        thread_rng().fill_bytes(&mut buf);
+        buf.to_vec()
+    }
+
+    #[test]
+    fn test_sign_and_verify() {
+        let key_manager = create_key_manager("test_output/rsa_sign_verify", None).unwrap();
+        let pem = key_manager.generate_rsa_keypair(&mut thread_rng()).unwrap();
```
Comment (danielemiliogarcia) [inline src/tests/rsa_test.rs:20]: I’d suggest renaming `pem` ((Privacy-Enhanced Mail)) to `pubkey_pem` for readability.
The method name `generate_rsa_keypair` suggests it returns a full keypair, so calling the result pem can be misleading to readers who aren’t familiar with the KeyManager API. Using pubkey_pem makes the intent explicit when verifying the signature.

the same apply for all the tests

**Review by danielemiliogarcia — APPROVED** [review]

File: src/tests/rsa_test.rs (hunk 1 above)
Comment (ValentinVaninetti) [reply src/tests/rsa_test.rs:20]: Thank you! will do the changes!
